### PR TITLE
Fix S6964 FP: Properties with default values

### LINKED ISSUE #9331: Fix S6964 FP: Property with a default value
### Description

[S6964](https://sonarsource.github.io/rspec/#/rspec/S6964/csharp) is reporting an error on properties with a default value

### Repro steps

```
public record Data
{
    public bool SomeProperty { get; init; } = false;
}

```

### Expected behavior

The rule is not reported as the property contains a default value

### Actual behavior

The rule is triggered and a warning is reported

### Known workarounds

ignore the rule

### Related information

* MSBuild / dotnet version
dotnet 8
* SonarScanner for .NET version (if used)
6.2.0
* Operating System
linux

### PR DESCRIPTION
**Pull request by zsolt-kolbay-sonarsource**
Fixes #9331

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs [hunk 1]
```diff
@@ -144,4 +146,11 @@ type switch
 
     private static bool HasValidateNeverAttribute(ISymbol symbol) =>
         symbol.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_ModelBinding_Validation_ValidateNeverAttribute);
+
+    private static bool HasDefaultValue(SyntaxNode node) => node switch
+    {
+        ParameterSyntax { Default: not null } => true,
+        PropertyDeclarationSyntax { Initializer: not null } property => property.Parent.ParameterList() is null,
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs:153]: Why `property.Parent.ParameterList() is null`? How are property initializers not relevant if there is a primary constructor?

File: analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs [hunk 2]
```diff
@@ -144,4 +146,11 @@ type switch
 
     private static bool HasValidateNeverAttribute(ISymbol symbol) =>
         symbol.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_ModelBinding_Validation_ValidateNeverAttribute);
+
+    private static bool HasDefaultValue(SyntaxNode node) => node switch
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs:150]: Slightly better name?
```suggestion
    private static bool IsInitialized(SyntaxNode node) => node switch
```

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Comment by mary-georgiou-sonarsource** [comment]
@zsolt-kolbay-sonarsource I removed my self since Martin already checked and approved!
I had couple of comments but Martin more or less covered them.

**Pull request merged**